---
title: "PreProt Description"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



## **General Description of PreProt**

PreProt is a graphical user interface (GUI) platform for the pre-processing of high-resolution labeled and label-free liquid chromatography-tandem mass spectrometry (LC-MS/MS) proteomics data sets. PreProt was developed using the shiny package, and different pre-processing methods, functions, and codes which were wrapped in one platform to harmonize the pre-processing of the proteomics data. PreProt inputs a MaxQuant “Proteingroup.txt” text file containing protein intensities to be pre-processed, performs different pre-processing steps and the final output is a pre-processed file ready for statistical and bioinformatics analysis.
PreProt pre-processing platform comprises a header section that hosts the title “PreProt”, the left main menu sidebar, and the main display on the right. The left main menu sidebar colored black hosts the six menus of the application and represents the various pre-processing stages of PreProt which also contain sub-menus. When a specific step on the menu is selected, the results of such an action are displayed on the right of PreProt’s body. PreProt application displays the results in form of data frames, plots, and prompt messages. The results from each step such as images and data frames can be downloaded using a download button at the bottom of the body display. Further, the user can navigate within and between different pages of the data frame using the page display at the bottom. When a dataset is uploaded to PreProt, the identified and quantified proteins are displayed in rows while the samples are displayed using columns. A specific protein can be searched using the search button on the top right of the PreProt platform as well as control the number of entries to be displayed by selecting the preferred number of entries where ten entries were set as the default.


## **Requirements**

### **1.Input files  requirements** 

PreProt accepts a tab-separated “ProteinGroups.txt” matrix file generated by MaxQuant software (Tyanova et al., 2016). MaxQuant outputs multiple tab-delimited text files that contained the data regarding the identified and quantified peptides, proteins, and their altered forms (Tyanova et al., 2016). The ProteinGroups.txt file is comprised of rows of the identified proteins and columns containing information related to the quantified proteins across the samples. These include Protein IDs, unique peptide counts, protein names, gene names, fasta headers, percentage of sequence coverage, intensity, corrected intensity, reverse sequences identified in the decoy database, and potential contaminants. The columns containing the identified protein intensities were the major focus in proteomics studies because they were used to extract relevant biological features and therefore required to be preprocessed to eliminate any technical variations. 
For TMT datasets, MaxQuant software also assigns a special identification identity to every sample in every batch. The unique identification is done on the protein intensity columns which are labeled as “Reporter intensity” and "Reporter intensity corrected" columns, where the special identification is assigned as two integers separated by an underscore character (_). The first integer represented the sample and the second integer represented the batch. In this case, "Reporter intensity corrected 1_1 represented the first sample in the first batch.  Additionally, an optional label dependent on the name assigned to the spectrum file before inputting it to the MaxQuant software could be found in between the unique integers assigned by MaxQuant. 
PreProt automatically recognizes all columns starting with the phrase/character string "Reporter intensity corrected" and “Reporter intensity” columns, followed by the unique identifiers (two integers separated by an underscore (_) or an optional label (characters only) between the integers. **It is therefore recommended that the proteinGroups.txt input file should maintain the columns of intensities with unique identifiers as two integers separated by an underscore (_) or an optional label (characters only) between the integers. For example, Reporter intensity corrected 1_1 or Reporter intensity corrected 1TMT1 for TMT datasets and "LFQ intensity" followed by an integer, for example, "LFQ intensity 01" for LFQ datasets.**

### **2. Metadata file requirements**

PreProt uses the function “plotvarpart()” To enable plotting a PVCA plot that helps to identify sources of variation in the data. The function, “plotvarpart()” requires a file (metadata) containing the available information about the experiment in columns. To accomplish the development of this stage in PreProt, a metadata file containing information about the experiment was required. A sample TMT metadata file was therefore prepared for the tool development which included the sample number, batch number, sex, sampling day, and analysis day (run day) which were gathered using standardized TMT experiment information (TMT sample data).  Additionally, extra variables case-control, sample type, sample id, and sample source commonly found in other proteomics experiments were added to the sample metadata file to enable PreProt to accommodate other proteomics datasets. 
The metadata file, **should**, therefore, contain the **sample number, sample id, batch number, sampling date, sex, case-control, sample type, sample source, and run day** in columns. The file is required to have the number of samples in rows which is expected to be equal to the number of samples in the data. It was therefore recommended that PreProt users curate their metadata files to have these variables incorporated (the column names) before uploading on PreProt. **However, if the metadata file was not available**, the PreProt application was designed to bypass the “variation inspection” stage if the file was not provided hence the user could directly go to inspect batch effect stage.
PreProt evaluates the possible sources of variation in the data using the sample metadata file and plots a principle variance component analysis (PVCA) using the function, “plotvarpart()” using the package variancePartition version 1.24.1 (Hoffman & Schadt, 2016). The y-axis on the PVCA plot explains the variance while the x-axis the possible sources of variations.  The plots are then replotted at the end after batch correction using the batch-corrected file to visualize the biological variation driving the data, the batch effects are expected to be removed. 


## **Detailed Preprocessing stages**


### **1. Data preparation**

Data preparation in PreProt is divided into LFQ and TMT data preparation stages where the user uploads the data files from the labeled (TMT) or label-free (LFQ) experiments.  A browse button to upload the data file to be pre-processed was incorporated in both the **“upload TMT raw data” and “upload LFQ raw data”** submenus of PreProt. Further, an additional submenu to filter the variables of interest using select buttons is incorporated in PreProt as **“filter preferred intensity columns” and “filter LFQ intensity column”** for TMT and LFQ, respectively. TMT datasets analyzed by MaxQuant contain two types of protein intensities (reporter intensity and reporter intensity corrected) and therefore PreProt incorporated an extra select menu in “filter preferred intensity columns” to enable the user to select the preferred intensities.  As PreProt allows analysis of LFQ and TMT data in one platform, a select button is incorporated to allow the selection of the type of analysis, where the “TMT” button was checked as the default analysis.

### **2. Data Cleaning**

The data cleaning step in PreProt has two submenus **“filter by threshold” and “filter contaminants”**. In the **“filter by threshold”** step, a slider button was incorporated to allow the selection of a threshold between 1 and 100 percent. By default, a threshold of 20% was set, which filters out the proteins not identified in 80% of the samples leaving only the proteins identified in more than 80% of the samples. The filtered content is displayed on the right side and the number of proteins can be seen reducing or increasing as the threshold is adjusted. 
In the **“filter contaminants”** step, a list of possible contaminants known to contaminate different proteomics experiments was incorporated to allow the selection and removal of such contaminants. The contaminants included were keratin, trypsin, lysine, amylase, albumin,  reverse sequences (REV), and proteins flagged with the word ‘CON’ during the process of protein identification and quantification in MaxQuant.

### **3. Imputation of missing values**

In PreProt, the imputation stage has three subsections namely; “inspect missing values”, “type of missing values”, and “choose imputation method”. The “Inspect missing values” provided a summary of missing values that were classified in frequencies of low (between 0-20% missingness), high (between 20-50-% missingness), and very high (between 50-100% missingness).
The “type of missing values”, in PreProt informed the user about the type of missing values in the data using a prompt message that indicated whether the data is **missing at random (MAR), completely at random (MCAR), or missing not at random (MNAR)**. Moreover, when informed about the type of missing values, PreProt allowed the user to choose a desirable imputation method while advising the user on the preferred method to use depending on the type of missingness. PreProt hosts four methods of imputation, namely;**K nearest neighbor (KNN) the default method, missRanger, missForest, and quantile regression imputation of the left censored data (QRLIC)**.

### **4. Normalization**

The normalization step in PreProt consists of three subsections namely, **“visualize before normalization”, “choose the normalization method”, and “visualize after normalization”**. The **“visualize before normalization”**, displayed boxplots of the input data before normalization. The second subsection enables data normalization, where select buttons containing four methods of normalization were incorporated. These included **variance stabilization normalization (VSN) the default method, Log normalization, Cyclic loess, and Quantile normalization method**. Ultimately, PreProt allows for the visualization of the data after normalization using boxplots using the submenu, **“visualize after normalization”**. Hence the performance of the normalization method can be judged by visualizing the data before and after normalization. 

### **5. Variation inspection and batch correction**

PreProt allows inspection of the possible sources of variation in the data using the identify sources of variation following an upload of the associated metadata using the **upload metadata** submenu. The associated metadata needs to be supplied by the user to enable examination of the main sources of the variation within the data. The first two subsections **“upload metadata” and “inspect sources of variation”** are optional as they aim to help identify how other related variables drive the variation in the data and rely on an external metadata file. However, the user can opt to directly go to the third subsection **“batch effects inspection”**if the metadata file is not available. 
The inspection of major sources of variations in the data was designed using a few common variables known to drive variation in proteomics data which are expected to be increased in the future versions of  PreProt. 
After loading the metadata file, the sources of variation were inspected using a principal variant component analysis (PVCA) plot followed by a PCA plot in the “visualize batch effect” submenu. The experimental batching once determined to cause variation in the data, PreProt performs batch correction using the combat algorithm and the batch-corrected file can be downloaded using the download button.
After batch correction, PreProt replots a PCA plot using the batch-corrected file to determine whether the batch effect are corrected. Finally, a PVCA plot is replotted to confirm the sources of variation in the data after batch correction which confirms whether the batch variation has been corrected.



